doc/AddingBlas3Op.md

```diff
@@ -0,0 +1,324 @@
+                                      ::testing::Values(1.0, 2.0),  // alpha
+                                      ::testing::Values(1.0, 2.0),  // lda_mul
+                                      ::testing::Values(1.0, 2.0),  // ldb_mul
+                                      ::testing::Values(0.0, NaN)   // unused
+);
+
+// Finaly register the new test
+BLAS_REGISTER_TEST(Trsm, combination_t, combi);
+```
+
+By this point, you are expected to get a linker error saying the symbol for the internal
+version of `blas::internal::_trsm` is undefined, and that is expected since we didn't 
+provide the internal implementation the newly added `_trsm` function.
+
+## Implement the internal version of `_trsm`
+
+The next step is to provide the implementation of `blas::internal::_trsm` and instantiate
+it somewhere.
+
+We start by creating the internal file `src/interface/trsm_interface.hpp`. This file
+will contain the actual implementation of `_trsm`, but note that the function is not
+yet being instantiated so the linker error will persist at this point:
+
+```c++
+namespace blas {
+namespace internal {
+
+template <typename executor_t, typename container_0_t, typename container_1_t,
+          typename element_t, typename index_t>
+typename executor_t::policy_t::event_t _trsm(
+    executor_t& ex, char Side, char Triangle, char Transpose, char Diagonal,
+    index_t M, index_t N, element_t alpha, container_0_t A, index_t lda,
+    container_1_t B, index_t ldb) {
+  // Implementation of the new operation
+  // This will probably invoke a kernel which we don't yet have defined.
+    
+  // This function can access any other operation defined in this
+  // or lower levels of blas. For example, one may want to invoke a gemm
+  // operation like the following
+  auto gemmEvent = internal::_gemm(
+            ex, 'n', isTranspose ? 't' : 'n', M, currentBlockSize,
+            currentBlockSize, (i == 0) ? alpha : element_t{1}, B + i * ldb, ldb,
+            invA + i * blockSize, blockSize, element_t{0}, X + i * ldx, ldx);
+  trsmEvents = concatenate_vectors(trsmEvents, gemmEvent);  
+
+  // Ultimately, a list of all events created in this function is returned
+  // to the user that can be used to wait on all or part of the internal
+  // operations required to perform the full computation
+  return trsmEvents;    
+}
+
+} // namespace internal
+} // namespace blas
+```
+
+## Instantiate the new operation
+
+The next step is to define the `_trsm` function in a `.cpp` file. In this way, when the client
+binary is linked against SYCL-BLAS, the linker will find the definition of the missing symbol.
+
+To do this, we create the source file that will contain instantiation of the new `_trsm` operation.
+The file is located at `src/interface/blas3/trsm.cpp.in`. This is not the file that will be 
+compiled, but a template file that the python script `python_generator/py_gen_blas_ternary.py`
```

Comment:
```suggestion
compiled, but a template file that the python script `python_generator/py_gen_blas_binary.py`
```

This should be binary if we are talking about TRSM right?

Reply:
Yes, thanks!